probe/data.go

```diff
@@ -93,13 +100,16 @@ func CleanData(p []Prober) {
 }
 
 // SaveDataToFile save the results to file
+// Note: we no need lock for this function, because this is only called once during the startup
 func SaveDataToFile(filename string) error {
 	metaData.file = filename
 	if strings.TrimSpace(filename) == "-" {
 		return nil
 	}
 
+	mutex.RLock()
```

Comment:
Conflict with the comment above

Reply:
My bad, remove the comment, this function needs lock